src/components/accueil/Accueil.tsx

```diff
@@ -59,6 +66,55 @@ export const Accueil = () => {
     return activitiesWithLastMimic;
   }
 
+  const getType = (typeValue: number): string | undefined => {
+    const type = Object.keys(ActivityType).find((key) => ActivityType[key] === typeValue);
+    return type;
+  };
+
+  const SUBTYPE_MAPPER = {
+    ENIGME: ENIGME,
+    INDICE: INDICE,
+    SYMBOL: SYMBOL,
+    DEFI: DEFI,
+    REPORTAGE: REPORTAGE,
+  };
+  const getSubtype = (typeName: string, subTypeValue: number): string | undefined => {
+    const result = Object.keys(SUBTYPE_MAPPER[typeName] || {}).find((key) => SUBTYPE_MAPPER[typeName][key] === subTypeValue);
+    return result;
+  };
+
+  function filterActivitiesByTerm(activitiesData: Activity<AnyData>[]): Activity<AnyData>[] {
+    const activitiesFilteredBySearchTermOnType = activitiesData.filter((activity) => {
+      const type = getType(activity.type);
+      const subType = getSubtype(type, activity.subType);
+      return subType?.toLowerCase().indexOf(filters.searchTerm.toLowerCase()) >= 0;
+    });
+    const activitiesFilteredBySearchTerm = activitiesData.filter((activity) => filterActvityByTerm(activity, filters.searchTerm));
+    let agregatedFilters = [...activitiesFilteredBySearchTerm, ...activitiesFilteredBySearchTermOnType];
+    agregatedFilters = [...new Set(agregatedFilters)];
+    return agregatedFilters;
+  }
```

Comment:
**Suggestion:** You can move these functions/constants outside the `Accueil` component. This will avoid defining them at each render. (The existing `filterActivitiesWithLastMimicGame` function should also be defined outside the component.)

Example:
```typescript
const getType = (typeValue: number): string | undefined => {
    [...]
};

const SUBTYPE_MAPPER = {
    ENIGME: ENIGME,
    INDICE: INDICE,
    SYMBOL: SYMBOL,
    DEFI: DEFI,
    REPORTAGE: REPORTAGE,
};

const getSubtype = (typeName: string, subTypeValue: number): string | undefined => {
    [...]
};

function filterActivityByTerm(activity: Activity, searchTerm: string) {
    [...]
}

function filterActivitiesByTerm(activities: Activity[], searchTerm: string): Activity<AnyData>[] {
    [...]
}

export const Accueil = () => {
    [...]
}
```